vulnerabilities/importers/openssl.py

```diff
@@ -20,101 +20,139 @@
 #  VulnerableCode is a free software code scanning tool from nexB Inc. and others.
 #  Visit https://github.com/nexB/vulnerablecode/ for support and download.
 
-import dataclasses
+import logging
 import re
 import xml.etree.ElementTree as ET
-from typing import Set
+from datetime import datetime
+from datetime import timezone
+from typing import Iterable
 
 import requests
 from packageurl import PackageURL
+from univers.version_range import OpensslVersionRange
+from univers.versions import OpensslVersion
 
-from vulnerabilities.helpers import create_etag
-from vulnerabilities.helpers import nearest_patched_package
-from vulnerabilities.importer import Advisory
+from vulnerabilities.importer import AdvisoryData
+from vulnerabilities.importer import AffectedPackage
 from vulnerabilities.importer import Importer
 from vulnerabilities.importer import Reference
+from vulnerabilities.importer import VulnerabilitySeverity
+from vulnerabilities.severity_systems import SCORING_SYSTEMS
 
+logger = logging.getLogger(__name__)
 
-class OpenSSLImporter(Importer):
 
+class OpensslImporter(Importer):
+    spdx_license_expression = "Apache-2.0"
+    license_url = "https://github.com/openssl/openssl/issues/17830"
     url = "https://www.openssl.org/news/vulnerabilities.xml"
 
-    def updated_advisories(self) -> Set[Advisory]:
-        # Etags are like hashes of web responses. We maintain
-        # (url, etag) mappings in the DB. `create_etag`  creates
-        # (url, etag) pair. If a (url, etag) already exists then the code
-        # skips processing the response further to avoid duplicate work
-        if create_etag(data_src=self, url=self.url, etag_key="ETag"):
-            raw_data = self.fetch()
-            advisories = self.to_advisories(raw_data)
-            return self.batch_advisories(advisories)
-
-        return []
-
-    def fetch(self):
-        return requests.get(self.url).content
-
-    @staticmethod
-    def to_advisories(xml_response: str) -> Set[Advisory]:
-        advisories = []
-        pkg_name = "openssl"
-        pkg_type = "generic"
+    def advisory_data(self) -> Iterable[AdvisoryData]:
+        xml_response = requests.get(self.url).content
```

Comment:
Do network calls in a separate function for simplicity of testing

Reply:
yeah, that makes sense